walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,120 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> availableTimeRanges = new ArrayList<>();
+    Set<String> attendees = new HashSet<>(request.getAttendees());
+    long duration = request.getDuration();
+
+    if (Long.compare(duration, TimeRange.WHOLE_DAY.duration()) > 0) {
+      // If the requested duration is longer than the {@code WHOLE_DAY}, there can't be any
```

Comment:
Remove the comments inside this if/else block. They're not adding anything useful that your code doesn't already make clear